core/src/sumeragi/main_loop.rs

```diff
@@ -935,9 +935,18 @@ pub(crate) fn run<F: FaultInjection>(
             // Checking if transactions are in the blockchain is costly
             .retain(|tx| !tx.is_expired(sumeragi.queue.tx_time_to_live));
 
-        sumeragi
-            .queue
-            .get_transactions_for_block(&state.wsv, &mut state.transaction_cache);
+        let mut expired_transactions = Vec::new();
+        sumeragi.queue.get_transactions_for_block(
+            &state.wsv,
+            &mut state.transaction_cache,
+            &mut expired_transactions,
+        );
+        sumeragi.send_events(
+            expired_transactions
+                .iter()
+                .map(Txn::expired_event)
+                .collect::<Vec<_>>(),
+        );
```

Comment:
I don't understand well the code in this file, but are you sure this is a good place to send events to the client? Shouldn't we record expired events into block as well as we do for failed events? And only then send this info?

Reply:
After some discussion in the DM we concluded that inclusion of `expired_transactions` is not necessary.